src/nimblepkg/sha1hashes.nim

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) Dominik Picheta. All rights reserved.
+# BSD License. Look at license.txt for more info.
+
+import strformat, strutils, json, std/sha1, hashes
+import common
+
+type
+  InvalidSha1HashError* = object of NimbleError
+    ## Represents an error caused by invalid value of a sha1 hash.
+
+  Sha1Hash* {.requiresInit.} = object
```

Comment:
I think that it is not possible, because they have not an appropriate constructor and validation. I want `Sha1Hash` to be created only with the `initSha1Hash` procedure. Here I'm using the same pattern as the case for `Version`.